python/whylogs/api/writer/whylabs.py

```diff
@@ -261,12 +263,21 @@ def __init__(
         pool = _UPLOAD_POOLER_CACHE.get(pooler_cache_key)
         if pool is None:
             logger.debug(f"Pooler is not available. Creating a new one for key: {pooler_cache_key}")
-            pool = PoolManager(
-                num_pools=4,
-                maxsize=10,
-                assert_hostname=self._s3_endpoint_subject,
-                server_hostname=self._s3_endpoint_subject,
-            )
+            if _https_proxy or _http_proxy:
+                pool = ProxyManager(
+                    _https_proxy or _http_proxy,
+                    num_pools=4,
+                    maxsize=10,
+                    assert_hostname=self._s3_endpoint_subject,
+                    server_hostname=self._s3_endpoint_subject,
+                )
+            else:
+                pool = PoolManager(
+                    num_pools=4,
+                    maxsize=10,
+                    assert_hostname=self._s3_endpoint_subject,
+                    server_hostname=self._s3_endpoint_subject,
+                )
             self._s3_pool = pool
             _UPLOAD_POOLER_CACHE[pooler_cache_key] = pool
```

Comment:
Do you know how the pooling is supposed to work? It looks like every instance of the whylabs writer has its own pool and there don't seem to be any concurrent requests (that are obvious). Are we actually getting benefit from the pool or are we just treating it like a single connection?

Reply:
I don't know what was the original motivation to use pooling. The scope of this PR was only to adapt the original code to accept proxy configurations through ProxyManager.

Reply:
We can easily saturate an endpoint by creating too many connections. The TLS handshake kills our upload FYI. This pooling enable us to re-use the TCP connection and don't have to repeatedly handshake with the endpoint